### PR TITLE
Making connection and stream windows configurable for Netty.

### PR DESCRIPTION
**Pull request by nmittler**

### CODE REVIEW

**Comment by nmittler** [comment]
@ejona86 can you take a look?

File: netty/src/main/java/io/grpc/transport/netty/NettyClientHandler.java [hunk 1]
```diff
@@ -446,6 +464,19 @@ private void removePendingStream(NettyClientStream stream) {
     return connection().activeStreams().toArray(new Http2Stream[0]);
   }
 
+  /**
+   * Initializes the connection window if we haven't already.
+   */
+  private void initConnectionWindow() throws Http2Exception {
+    if (connectionWindowSize > 0 && ctx.channel().isActive()) {
```
Comment (ejona86) [inline netty/src/main/java/io/grpc/transport/netty/NettyClientHandler.java]: Isn't connectionWindowSize always > 0 (due to Precondition)?

Comment (ejona86) [inline netty/src/main/java/io/grpc/transport/netty/NettyClientHandler.java]: Nevermind. I see.

**Comment by ejona86** [comment]
@nmittler, Travis is failing (and I re-ran it, and same unknown error), but otherwise LGTM

**Comment by nmittler** [comment]
cherry-picked into master.

**Pull request closed without merging**